apps/relayer/src/transaction/transaction.service.ts

```diff
@@ -113,24 +124,6 @@ export class TransactionService implements OnModuleInit, OnModuleDestroy {
   private watchTransaction(txHash: string, result?: TransactionResult) {
     this.watchedTransactions.add(txHash)
     this.transactionSeenAt.set(txHash, Date.now())
-
-    if (result !== undefined) {
-      this.waitForReceipt(result).catch(e => {
```

Comment:
Yeah, I dunno why I insisted on using the receipt here and not just relied on the overall polling 😅
Also I guess you can remove the optional `result` argument of the function.

Reply:
Yeah, I think it's a bit cleaner to just rely on a single mechanism to update the state. Good point on the `result` field.. removed!